### PR TITLE
Added an ability to use null values for iaq in the sensors

### PR DESCRIPTION
**Pull request by grkek**

### CODE REVIEW

File: drivers/vergesense/location_service.cr [hunk 1]
```diff
@@ -212,7 +212,8 @@ class Vergesense::LocationService < PlaceOS::Driver
             when .air_quality?
               id = "air"
               time = space.environment.try &.timestamp
-              space.environment.try &.iaq.value
+              iaq = space.environment.try &.iaq
+              iaq.nil? ? nil : iaq.not_nil!.value
```
Comment (caspiano) [inline drivers/vergesense/location_service.cr:216]:
```suggestion
              space.environment.try(&.iaq.try(&.value))
```

File: drivers/vergesense/room_sensor.cr [hunk 2]
```diff
@@ -62,9 +62,10 @@ class Vergesense::RoomSensor < PlaceOS::Driver
       self[:people] = 0
     end
 
+    iaq = floor_space.environment.try &.iaq
```
Comment (caspiano) [inline drivers/vergesense/room_sensor.cr:65]:
```suggestion
```

File: drivers/vergesense/room_sensor.cr [hunk 3]
```diff
@@ -62,9 +62,10 @@ class Vergesense::RoomSensor < PlaceOS::Driver
...
     self[:humidity] = floor_space.environment.try &.humidity.value
     self[:temperature] = floor_space.environment.try &.temperature.value
-    self[:air_quality] = floor_space.environment.try &.iaq.value
+    self[:air_quality] = iaq.nil? ? nil : iaq.not_nil!.value
```
Comment (caspiano) [inline drivers/vergesense/room_sensor.cr:68]:
```suggestion
    self[:air_quality] = floor_space.environment.try(&.iaq.try(&.value))
```

Not ideal, but maybe it's worth allowing the other sensors to be nil, and creating a helper that can warn when an integration is not returning values. Suggestion below might be a bit overkill

```crystal
module Vergesense
  struct Environment
    {% begin %}
    {% for reading in [iaq, temperature, humidity] %}
      getter {{ reading }} : Sensor?
      
      def read_{{ reading }}
        if reading = {{ reading }}
          reading.value
        else
          Log.warn { "missing reading for {{ reading }}" }
          nil
        end
      end
    {% end %}
    {% end %}
  end
end
```

File: drivers/vergesense/room_sensor.cr [hunk 4]
```diff
@@ -136,7 +137,8 @@ class Vergesense::RoomSensor < PlaceOS::Driver
             when .air_quality?
               id = "air_quality"
               time = space.environment.try &.timestamp
-              space.environment.try &.iaq.value
+              iaq = space.environment.try &.iaq
+              iaq.nil? ? nil : iaq.not_nil!.value
```
Comment (caspiano) [inline drivers/vergesense/room_sensor.cr:141]: Ditto suggestion as previous iag access.

**Review by caspiano — CHANGES_REQUESTED** [review]

File: drivers/vergesense/room_sensor.cr (hunk 4 above)
Comment (stakach) [reply drivers/vergesense/room_sensor.cr:141]:
```suggestion
              space.environment.try(&.iaq.try(&.value))
```

**Pull request merged**